pkg/customresourcestate/registry_factory.go

```diff
@@ -671,22 +707,95 @@ func famGen(f compiledFamily) generator.FamilyGenerator {
 	}
 }
 
+func resolveWildcard(path valuePath, object map[string]interface{}) []valuePath {
+	if path == nil {
+		return nil
+	}
+	fn := func(i *int) bool {
```

Comment:
Can we move this function outside and give it a better name like "findWildcard" or similar?